prow/plugins/milestoneapplier/milestoneapplier_test.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package milestoneapplier
+
+import (
+	"testing"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+)
+
+func TestMilestoneApplier(t *testing.T) {
+	var milestonesMap = map[string]int{"v1.0": 1, "v2.0": 2}
+	testcases := []struct {
+		name                string
+		baseBranch          string
+		prAction            github.PullRequestEventAction
+		merged              bool
+		previousMilestone   int
+		configuredMilestone int
+		expectedMilestone   int
+	}{
+		{
+			name:              "opened PR on default branch => do nothing",
+			baseBranch:        "master",
+			prAction:          github.PullRequestActionOpened,
+			expectedMilestone: 0,
+		},
+		{
+			name:              "closed (not merged) PR on default branch => do nothing",
+			baseBranch:        "master",
+			prAction:          github.PullRequestActionClosed,
+			merged:            false,
+			expectedMilestone: 0,
+		},
+		{
+			name:                "merged PR but has existing milestone on default branch => do nothing",
```

Comment:
I won't enumerate the rest of the test cases, but this is the behavior I think should change.  I would differentiate between "wrong existing milestone" and "correct existing milestone".  This looks like the "wrong" case and I would expect `expectedMilestone` to be `2` (matching `configuredMilestone`)